pkg/controller/stats_socket.go

```diff
@@ -0,0 +1,189 @@
+
+							toRemoveEndpoints := endpointsSubtract(curEndpoints, updEndpoints)
+							toAddEndpoints := endpointsSubtract(updEndpoints, curEndpoints)
+
+							// check for new/removed entries in this backend, issue socket commands
+							backendSlots := updatedConfig.BackendSlots[backendName]
+							// remove endpoints
+							for k := range toRemoveEndpoints {
+								err := utils.SendToSocket(haproxy.CurrentConfig.Cfg.StatsSocket,
+									fmt.Sprintf("set server %s/%s state maint\n", backendName, backendSlots.FullSlots[k].BackendServerName))
+								if err != nil {
+									glog.Warningln("failed socket command srv remove")
+									haproxy.ReloadRequired = true
+								}
+								backendSlots.EmptySlots = append(backendSlots.EmptySlots, backendSlots.FullSlots[k].BackendServerName)
+								delete(backendSlots.FullSlots, k)
+							}
+							sort.Strings(backendSlots.EmptySlots)
+							// add endpoints
+							for k, endpoint := range toAddEndpoints {
+								// rearrange slots
+								backendSlots.FullSlots[k] = types.HAProxyBackendSlot{
+									//backendSlots.EmptySlots[len(backendSlots.EmptySlots)-1],
+									backendSlots.EmptySlots[0],
+									endpoint,
+								}
+								backendSlots.EmptySlots = backendSlots.EmptySlots[1:]
+
+								// send socket commands
+								err1 := utils.SendToSocket(haproxy.CurrentConfig.Cfg.StatsSocket,
+									fmt.Sprintf("set server %s/%s addr %s port %s\n", backendName, backendSlots.FullSlots[k].BackendServerName, endpoint.Address, endpoint.Port))
+								err2 := utils.SendToSocket(haproxy.CurrentConfig.Cfg.StatsSocket,
+									fmt.Sprintf("set server %s/%s state ready\n", backendName, backendSlots.FullSlots[k].BackendServerName))
+								if err1 != nil || err2 != nil {
+									glog.Warningln("failed socket command srv add")
+									haproxy.ReloadRequired = true
+								}
+							}
+							updatedConfig.BackendSlots[backendName] = backendSlots
+							// reload if any socket commands were unsuccessful
+						}
+					}
+				}
+			}
+			// restore backend lists
+			haproxy.CurrentConfig.Backends = curBackends
+			updatedConfig.Backends = updBackends
+			// restore backend slots
+			haproxy.CurrentConfig.BackendSlots = curBackendSlots
+		}
+	}
+
+	// fill-out backends with available endpoints, add empty slots if required
+	if haproxy.CurrentConfig == nil || reconfigureEmptySlots {
+		haproxy.ReloadRequired = true
+		updBackendsMap, updKeys := ingressBackendsRemap(updatedConfig.Backends)
+		updatedConfig.BackendSlots = map[string]types.HAProxyBackendSlots{}
+
+		for _, backendName := range updKeys {
+			newBackend := types.HAProxyBackendSlots{}
+			newBackend.FullSlots = map[string]types.HAProxyBackendSlot{}
+			for i, endpoint := range updBackendsMap[backendName].Endpoints {
+				newBackend.FullSlots[fmt.Sprintf("%s:%s", endpoint.Address, endpoint.Port)] = types.HAProxyBackendSlot{
+					fmt.Sprintf("server%04d", i),
+					&endpoint,
+				}
```

Comment:
It is possible to use `addr:port` if `DynamicScaling` is false, I'll change to that. You are correct that it's not feasible if `DynamicScaling` is enabled.